File: ios/App/App/MyCapWebview.m
//
//  MyCapWebview.m
//  App
//
//  Created by Sambit on 07/05/20.
//

@import Capacitor;
@import Foundation;

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(MyCapWebview, "MyCapWebview",
           CAP_PLUGIN_METHOD(loadApp, CAPPluginReturnPromise);
)

File: ios/App/App/MyCapWebview.swift
//
//  MyWebview.swift
//  App
//
//  Created by Sambit on 06/05/20.
//

import UIKit
import Capacitor

@objc(MyCapWebview) public class MyCapWebview: CAPPlugin, WKNavigationDelegate {
    private var oldDelegate: WKNavigationDelegate?
    @objc public func loadApp(_ call: CAPPluginCall) {
        let webapp = call.options["webapp"]!
        let capViewController = bridge.viewController as! CAPBridgeViewController
        var path = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first!
        path = "\(path)/\(webapp)"
        capViewController.setServerBasePath(path: path)
        call.resolve()
    }
}
